src/package.rs

```diff
@@ -159,32 +250,94 @@ impl PackageStore {
             .wrap_err("Failed to release package")?
             .into();
 
-        tracing::info!("+ packaged {}@{}", api.name, api.version);
+        tracing::info!(":: packaged {}@{}", pkg.name, pkg.version);
 
-        Ok(Package {
-            name: api.name,
-            version: api.version,
-            tgz,
-        })
+        Ok(Package::new(pkg, tgz))
     }
 }
 
 /// An in memory representation of a `buffrs` package
 #[derive(Clone, Debug, Hash, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Package {
-    /// The name of the package
-    pub name: PackageId,
-    /// The version of the package
-    pub version: String,
+    /// Manifest of the package
+    pub manifest: PackageManifest,
     /// The `tar.gz` archive containing the protocol buffers
     #[serde(skip)]
     pub tgz: Bytes,
 }
 
 impl Package {
     /// Creates a new package
-    pub fn new(name: PackageId, version: String, tgz: Bytes) -> Self {
-        Self { name, version, tgz }
+    pub fn new(manifest: PackageManifest, tgz: Bytes) -> Self {
+        Self { manifest, tgz }
+    }
+
+    /// Decodes the tar and returns the encoded manifest
+    pub fn decode(tgz: Bytes) -> eyre::Result<Self> {
```

Comment:
`impl TryFrom<Bytes>`?

Reply:
true!

but the weird thing here is that we only do some introspection and then still store the compressed tgz

But i agree when seeing this a second time, it should be try from